storage/remote/write_handler_test.go

```diff
@@ -165,6 +167,62 @@ func TestCommitErr(t *testing.T) {
 	require.Equal(t, "commit error\n", string(body))
 }
 
+func BenchmarkRemoteWriteOOOSamples(b *testing.B) {
```

Comment:
I think this is not the right place to benchmark this. We want to benchmark only the ingestion part for the chunk, and it is best to have it as close to where it is happening. Other things like building the request, serving it, etc, can shadow the other numbers.

Reply:
Had a chat offline with Jesus. This place makes sense.

Reply:
For clarification, the reason why it makes sense is because we typically receive ooo samples through remote write and not through scrapes. So a benchmark here makes sense from the perspective of how prometheus receives ooo samples.

> Other things like building the request, serving it, etc, can shadow the other numbers.

You are right, this is why I'm using `b.ResetTimer()` at the right place to avoid shadowing what we're actually benchmarking here.

Reply:
I meant the lines inside the loop after b.ResetTimer, until you call ServeHTTP, could add extra overhead for every iteration. But if it is faster even after including that, then it's good!